src/components/table/table.html

```diff
@@ -1,34 +1,81 @@
-<table class="m-table" :class="[ 'm--is-skin-' + skin]" cellspacing="0">
-    <slot name="header">
-        <thead>
-            <tr>
-                <slot v-for="column in columns" :name="'header.' + column.dataProp">
-                    <th :style="columnWidth(column)" :key="column.id">{{ column.title }}</th>
-                </slot>
-            </tr>
-        </thead>
-    </slot>
+<div class="m-table">
+    <div class="m-table__header"
+         v-if="hasHeader"
+         ref="header">
+        <h2 class="m-u--no-margin"
+            v-if="title">{{ title }}</h2>
+        <div class="m-table__header__spacer"></div>
+        <slot name="search"></slot>
+        <m-link class="m-table__header__add"
+                :class="{'m--has-margin': $slots.search}"
+                v-if="addBtnLabel"
+                :icon="true"
+                icon-name="m-svg__add-circle-filled"
+                icon-size="22px"
+                mode="button"
+                @click="add"
+                ref="addBtn">{{ addBtnLabel }}</m-link>
+    </div>
 
-    <slot name="body">
-        <tbody>
-            <tr v-if="isEmpty">
-                <slot name="empty">
-                    <td class="m-table--empty" :colspan="columns.length">
-                        <p>{{ i18nEmptyTable }}</p>
-                    </td>
-                </slot>
-            </tr>
-            <tr v-else v-for="(row, index) in rows" :key="index">
-                <slot v-for="column in columns" :name="'body.' + column.dataProp" :data="row">
-                    <td :key="'body.' + column.id" >{{ row[column.dataProp] }}</td>
-                </slot>
-            </tr>
-        </tbody>
-    </slot>
+    <table class="m-table__table"
+           :class="['m--is-skin-' + skin, {'m--is-loading': loading}]"
+           cellspacing="0">
+        <slot name="header">
+            <thead>
+                <tr>
+                    <slot v-for="column in columns"
```

Comment:
je pense que c'était un oubli initial mais il faut toujours ajouter :key au boucle v-for

Reply:
Vue génère une erreur si on met la prop key sur une slot : (Emitted value instead of an instance of Error) template syntax error `key` does not work on <slot> because slots are abstract outlets and can possibly expand into multiple elements. Use the key on a wrapping element instead.

Reply:
ish c'est problèmatique ca, ça na rien a voir avec ton PR donc je vais traiter ça dans une autre demande.  

Il faudra inverser les tag <slot> avec le tag <td>  meme chose pour le <th> Car si on ne met pas de :key dans le for et que les personne ne le mettre pas dans leur <template> ** comme c'est le cas dans la sandbox du complex table actuellement on va avoir des problemes de performance pour les gros tableau...on s'en reparle! je vais accepter les changements dans ce PR

Reply:
D'ailleurs parlant de performance => Dans tous mes jobs précédent dans notre page de démo / test/ sandbox de tableaux on avait toujours un test de perf avec quelques milliers d'enregistrement.  Ça va toujours finir par ralentir, mais on voulait garder un oeil dessus en permanance.

Serait donc intéressant d'ajouter ce test éventuellement

Reply:
@Mboulianne je metterais pas ça directement les sandboxes mais c'est clair que ça prend des tests de performance, a voir ou et comment on pourrait les faires

Reply:
Ouep tant qu'ils sont à quelque part.
En passant Carl, tu as pas besoin de tenir compte de mon commentaire pour ta PR :P.  En principe je suis pas reviewer modUL cette semaine :'D